internal/clients/creds_cache.go

```diff
@@ -0,0 +1,202 @@
+// SPDX-FileCopyrightText: 2024 The Crossplane Authors <https://crossplane.io>
+//
+// SPDX-License-Identifier: Apache-2.0
+
+package clients
+
+import (
+	"context"
+	"crypto/sha256"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/crossplane/crossplane-runtime/pkg/logging"
+	"github.com/pkg/errors"
+	"sigs.k8s.io/controller-runtime/pkg/log/zap"
+
+	"github.com/upbound/provider-aws/apis/v1beta1"
+)
+
+// GlobalAWSCredentialsProviderCache is a global AWS CredentialsProvider cache to be used by all controllers.
+var GlobalAWSCredentialsProviderCache = NewAWSCredentialsProviderCache()
```

Comment:
We may consider getting rid of this global cache variable by initializing a cache in `clients.SelectTerraformSetup`.